search/src/App.jsx

```diff
@@ -26,31 +26,40 @@ class App extends Component {
       items: [],
       loggedIn: null,
       err: null,
+      link: '#'
     }
 
     this.masq = null
     this.onSearch = this.onSearch.bind(this)
     this.handleClickLogin = this.handleClickLogin.bind(this)
     this.handleClickLogout = this.handleClickLogout.bind(this)
+    this.getAllQueriesFromDB = this.getAllQueriesFromDB.bind(this)
+  }
+
+  getAllQueriesFromDB () {
+    const db = this.masq._getDB()
+
+    // Retrieve existing keys, in order to
+    // retrieve items from the values stored in the DB
+    db.list((err, list) => {
+      if (err) console.error(err)
+      const items = list.map(nodes => nodes[0].key)
+      this.setState({ items })
+    })
   }
 
   async componentDidMount () {
     this.masq = new Masq(APP.name, APP.description, APP.imageURL)
+
+    const { link } = await this.masq.logIntoMasq(true)
+    this.setState({ link })
+
     const loggedIn = this.masq.isLoggedIn()
     if (loggedIn) {
```

Comment:
If we test whether we are logged in, we can also test if this.masq.isConnected() is true